### PR TITLE
add support for a 4th param in v2, reset chain

### PR DESCRIPTION
**Pull request by maxtaco**
- reset chain is [seqno, hash]

### CODE REVIEW

File: test/files/30_merkle_leaf.iced [hunk 1]
```diff
@@ -45,19 +44,25 @@ exports.test_v2_3 = (T,cb) ->
...
     T.assert err?, "parse error on object #{i}"
 
-  goods = [ 
+  goods = [
     [ 2, [1, "", "" ]  ],
     [ 2, [], [], "aa" ]
     [ 2, null, null, "aa" ]
     [ 2, [], null, "aa" ]
     [ 2, null, [], "aa" ]
+    [ 2, [10, "aa", "bb"], null, null ]
```
Comment (oconnor663) [inline test/files/30_merkle_leaf.iced]: What is this case checking for?

File: src/merkle/leaf.iced [hunk 2]
```diff
@@ -9,6 +9,20 @@ exports.ChainTail = class ChainTail
...
+  constructor : ({@seqno, @payload_hash}) ->
+  to_json : () -> [ @seqno, @payload_hash ]
+
+#--------------------------
+
+is_positive_int = (x) ->
+  return false unless typeof(x) is 'number'
+  return false unless Math.floor(x) is x
+  return false if x < 0
+  return true
```
Comment (oconnor663) [inline src/merkle/leaf.iced]: Can we use the `Number.isInteger` function? Or maybe [the polyfill suggested](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger) in that function's docs?

File: test/files/30_merkle_leaf.iced (hunk 1 above)
Comment (maxtaco) [reply test/files/30_merkle_leaf.iced]: a `null` eldest which should be OK

File: src/merkle/leaf.iced (hunk 2 above)
Comment (maxtaco) [reply src/merkle/leaf.iced]: 👍

**Pull request merged**